pkg/sbom/spdx/marshal_test.go

```diff
@@ -0,0 +1,674 @@
+						PackageVersion:          "2.2.53",
+						PackageLicenseConcluded: "GPLv2+",
+						PackageLicenseDeclared:  "GPLv2+",
+						PackageExternalReferences: []*spdx.PackageExternalReference2_2{
+							{
+								Category: tspdx.CategoryPackageManager,
+								RefType:  tspdx.RefTypePurl,
+								Locator:  "pkg:rpm/centos/acl@1:2.2.53-1.el8?arch=aarch64&distro=centos-8.3.2011",
+							},
+						},
+						PackageSourceInfo: "built package from: 1:acl-2.2.53-1.el8",
+					},
+					spdx.ElementID("Package-826226d056ff30c0"): {
+						PackageSPDXIdentifier:   spdx.ElementID("Package-826226d056ff30c0"),
+						PackageName:             "actionpack",
+						PackageVersion:          "7.0.1",
+						PackageLicenseConcluded: "NONE",
+						PackageLicenseDeclared:  "NONE",
+						PackageExternalReferences: []*spdx.PackageExternalReference2_2{
+							{
+								Category: tspdx.CategoryPackageManager,
+								RefType:  tspdx.RefTypePurl,
+								Locator:  "pkg:gem/actionpack@7.0.1",
+							},
+						},
+						PackageAttributionTexts: []string{
+							"LayerDiffID: sha256:ccb64cf0b7ba2e50741d0b64cae324eb5de3b1e2f580bbf177e721b67df38488",
+						},
+						Files: map[spdx.ElementID]*spdx.File2_2{
+							"File-fa42187221d0d0a8": {
+								FileSPDXIdentifier: "File-fa42187221d0d0a8",
+								FileName:           "tools/project-doe/specifications/actionpack.gemspec",
+							},
+							"File-6a540784b0dc6d55": {
+								FileName:           "tools/project-john/specifications/actionpack.gemspec",
+								FileSPDXIdentifier: "File-6a540784b0dc6d55",
+							},
+						},
+					},
+				},
+				Relationships: []*spdx.Relationship2_2{
+					{
+						RefA:         spdx.DocElementID{ElementRefID: "DOCUMENT"},
+						RefB:         spdx.DocElementID{ElementRefID: "RootPackage-413bfede37ad01fc"},
+						Relationship: "DESCRIBE",
+					},
+					{
+						RefA:         spdx.DocElementID{ElementRefID: "RootPackage-413bfede37ad01fc"},
+						RefB:         spdx.DocElementID{ElementRefID: "OperatingSystem-197f9a00ebcb51f0"},
+						Relationship: "CONTAINS",
+					},
+					{
+						RefA:         spdx.DocElementID{ElementRefID: "OperatingSystem-197f9a00ebcb51f0"},
+						RefB:         spdx.DocElementID{ElementRefID: "Package-d8dccb186bafaf37"},
+						Relationship: "DEPENDS_ON",
+					},
+					{
+						RefA:         spdx.DocElementID{ElementRefID: "RootPackage-413bfede37ad01fc"},
+						RefB:         spdx.DocElementID{ElementRefID: "Application-441a648f2aeeee72"},
+						Relationship: "CONTAINS",
+					},
+					{
+						RefA:         spdx.DocElementID{ElementRefID: "Application-441a648f2aeeee72"},
+						RefB:         spdx.DocElementID{ElementRefID: "Package-826226d056ff30c0"},
+						Relationship: "DEPENDS_ON",
+					},
+					{
+						RefA:         spdx.DocElementID{ElementRefID: "Application-441a648f2aeeee72"},
+						RefB:         spdx.DocElementID{ElementRefID: "Package-826226d056ff30c0"},
+						Relationship: "DEPENDS_ON",
+					},
```

Comment:
The logic for generating the ID of a package has been designed to ignore the FilePath, so the relation is duplicated as the same package.
I would like to discuss whether packages with different FilePaths should be treated as identical.